src/common/buffer.cc

```diff
@@ -41,7 +41,7 @@
 
 using namespace ceph;
 
-#define CEPH_BUFFER_ALLOC_UNIT  (std::min(CEPH_PAGE_SIZE, 4096u))
+#define CEPH_BUFFER_ALLOC_UNIT  4096u
```

Comment:
i think it's a safe change. but could you reference some specs prove that 4k is the minimal pagesize on all architectures we support?

for instance, 

- https://github.com/torvalds/linux/blob/6f0d349d922ba44e4348a17a78ea51b7135965b1/include/asm-generic/page.h#L16
- https://github.com/torvalds/linux/blob/master/arch/arm/include/asm/page.h#L15
- https://github.com/torvalds/linux/blob/e2b623fbe6a34bce1332584212ae101ebc2508f5/arch/s390/include/asm/page.h#L14

and probably *Intel® 64 and IA-32 Architectures Software Developer’s Manual*, Vol3, table 4-1, Properties of Different Paging Modes, https://software.intel.com/sites/default/files/managed/a4/60/325384-sdm-vol-3abcd.pdf .

Reply:
If we follow the assumption that Ceph requires Linux, then bulky `greps` for `PAGE_SIZE` might be the way. Let's try:
```
/work/linux$ grep -r -E '#define[[:space:]]*PAGE_SIZE[[:space:]]' ./arch/
./arch/arm64/include/asm/page-def.h:#define PAGE_SIZE		(_AC(1, UL) << PAGE_SHIFT)
./arch/microblaze/include/asm/page.h:#define PAGE_SIZE	(ASM_CONST(1) << PAGE_SHIFT)
./arch/um/include/asm/page.h:#define PAGE_SIZE	(_AC(1, UL) << PAGE_SHIFT)
./arch/s390/include/asm/page.h:#define PAGE_SIZE	_PAGE_SIZE
./arch/hexagon/kernel/vmlinux.lds.S:#define PAGE_SIZE _PAGE_SIZE
./arch/hexagon/include/asm/page.h:#define PAGE_SIZE  (1UL << PAGE_SHIFT)
./arch/powerpc/boot/page.h:#define PAGE_SIZE	(ASM_CONST(1) << PAGE_SHIFT)
./arch/powerpc/include/asm/page.h:#define PAGE_SIZE		(ASM_CONST(1) << PAGE_SHIFT)
./arch/x86/include/asm/page_types.h:#define PAGE_SIZE		(_AC(1,UL) << PAGE_SHIFT)
./arch/arm/include/asm/page.h:#define PAGE_SIZE		(_AC(1,UL) << PAGE_SHIFT)
./arch/parisc/include/asm/page.h:#define PAGE_SIZE	(_AC(1,UL) << PAGE_SHIFT)
./arch/sparc/include/asm/page_64.h:#define PAGE_SIZE    (_AC(1,UL) << PAGE_SHIFT)
./arch/sparc/include/asm/page_32.h:#define PAGE_SIZE    (_AC(1, UL) << PAGE_SHIFT)
./arch/m68k/include/asm/page.h:#define PAGE_SIZE	(_AC(1, UL) << PAGE_SHIFT)
./arch/openrisc/include/asm/page.h:#define PAGE_SIZE       (1 << PAGE_SHIFT)
./arch/openrisc/include/asm/page.h:#define PAGE_SIZE       (1UL << PAGE_SHIFT)
./arch/arc/include/uapi/asm/page.h:#define PAGE_SIZE	_BITUL(PAGE_SHIFT)	/* Default 8K */
./arch/nios2/include/asm/page.h:#define PAGE_SIZE	(_AC(1, UL) << PAGE_SHIFT)
./arch/unicore32/include/asm/page.h:#define PAGE_SIZE		(_AC(1, UL) << PAGE_SHIFT)
./arch/alpha/include/asm/page.h:#define PAGE_SIZE	(_AC(1,UL) << PAGE_SHIFT)
./arch/xtensa/include/asm/page.h:#define PAGE_SIZE	(__XTENSA_UL_CONST(1) << PAGE_SHIFT)
./arch/ia64/include/asm/page.h:#define PAGE_SIZE		(__IA64_UL_CONST(1) << PAGE_SHIFT)
./arch/mips/kernel/vmlinux.lds.S:#define PAGE_SIZE _PAGE_SIZE
./arch/mips/include/asm/page.h:#define PAGE_SIZE	(_AC(1,UL) << PAGE_SHIFT)
./arch/riscv/include/asm/page.h:#define PAGE_SIZE	(_AC(1, UL) << PAGE_SHIFT)
./arch/nds32/include/asm/page.h:#define PAGE_SIZE       (_AC(1,UL) << PAGE_SHIFT)
./arch/sh/include/asm/page.h:#define PAGE_SIZE	(_AC(1, UL) << PAGE_SHIFT)
```
These values usually depend on `PAGE_SHIFT`.
```
/work/linux$ grep -r -E '#define[[:space:]]*PAGE_SHIFT[[:space:]]' ./arch/
./arch/arm64/include/asm/page-def.h:#define PAGE_SHIFT		CONFIG_ARM64_PAGE_SHIFT
./arch/microblaze/include/asm/page.h:#define PAGE_SHIFT		16
./arch/microblaze/include/asm/page.h:#define PAGE_SHIFT		14
./arch/microblaze/include/asm/page.h:#define PAGE_SHIFT		12
./arch/um/include/asm/page.h:#define PAGE_SHIFT	12
./arch/s390/include/asm/page.h:#define PAGE_SHIFT	_PAGE_SHIFT
./arch/hexagon/include/asm/page.h:#define PAGE_SHIFT 12
./arch/hexagon/include/asm/page.h:#define PAGE_SHIFT 14
./arch/hexagon/include/asm/page.h:#define PAGE_SHIFT 16
./arch/hexagon/include/asm/page.h:#define PAGE_SHIFT 18
./arch/hexagon/include/asm/page.h:#define PAGE_SHIFT 20
./arch/powerpc/boot/page.h:#define PAGE_SHIFT	12
./arch/powerpc/include/asm/page.h:#define PAGE_SHIFT		18
./arch/powerpc/include/asm/page.h:#define PAGE_SHIFT		16
./arch/powerpc/include/asm/page.h:#define PAGE_SHIFT		14
./arch/powerpc/include/asm/page.h:#define PAGE_SHIFT		12
./arch/x86/include/asm/page_types.h:#define PAGE_SHIFT		12
./arch/arm/include/asm/page.h:#define PAGE_SHIFT		12
./arch/sparc/include/asm/page_64.h:#define PAGE_SHIFT   13
./arch/sparc/include/asm/page_32.h:#define PAGE_SHIFT   12
./arch/m68k/include/asm/page.h:#define PAGE_SHIFT	13
./arch/m68k/include/asm/page.h:#define PAGE_SHIFT	12
./arch/openrisc/include/asm/page.h:#define PAGE_SHIFT      13
./arch/arc/include/uapi/asm/page.h:#define PAGE_SHIFT 14
./arch/arc/include/uapi/asm/page.h:#define PAGE_SHIFT 12
./arch/arc/include/uapi/asm/page.h:#define PAGE_SHIFT 13
./arch/nios2/include/asm/page.h:#define PAGE_SHIFT	12
./arch/unicore32/include/asm/page.h:#define PAGE_SHIFT		12
./arch/alpha/include/asm/page.h:#define PAGE_SHIFT	13
./arch/xtensa/include/asm/page.h:#define PAGE_SHIFT	12
./arch/mips/include/asm/page.h:#define PAGE_SHIFT	12
./arch/mips/include/asm/page.h:#define PAGE_SHIFT	13
./arch/mips/include/asm/page.h:#define PAGE_SHIFT	14
./arch/mips/include/asm/page.h:#define PAGE_SHIFT	15
./arch/mips/include/asm/page.h:#define PAGE_SHIFT	16
./arch/riscv/include/asm/page.h:#define PAGE_SHIFT	(12)
./arch/nds32/include/asm/page.h:#define PAGE_SHIFT      12
./arch/nds32/include/asm/page.h:#define PAGE_SHIFT      13
```

```
/work/linux$ grep -A5 ARM64_PAGE_SHIFT arch/arm64/Kconfig
config ARM64_PAGE_SHIFT
	int
	default 16 if ARM64_64K_PAGES
	default 14 if ARM64_16K_PAGES
	default 12
```

```
$ grep -r -E '#define[[:space:]]*_PAGE_SIZE[[:space:]]' ./arch/
./arch/s390/include/asm/page.h:#define _PAGE_SIZE	(_AC(1, UL) << _PAGE_SHIFT)
$ grep -r -E '#define[[:space:]]*_PAGE_SHIFT[[:space:]]' ./arch/
./arch/s390/include/asm/page.h:#define _PAGE_SHIFT	12
```

Reply:
@rzarzynski i agree that 4k is the minimal pagesize used by Linux. and what i wanted is to have a more convincing proof in the commit message than

> > It sounds extremely unlikely that any platform Ceph targets can have
PAGE_SIZE lower than 4k.

probably https://en.wikipedia.org/wiki/Page_%28computer_memory%29 would suffice. and snippet of your grep research would help also. but it is a little bit lengthy.